pkg/network/protocols/kafka/stats.go

```diff
@@ -65,12 +64,20 @@ type RequestStat struct {
 	StaticTags         uint64
 }
 
-func (r *RequestStat) initSketch() (err error) {
-	r.Latencies, err = ddsketch.NewDefaultDDSketch(RelativeAccuracy)
-	if err != nil {
-		log.Debugf("error recording kafka transaction latency: could not create new ddsketch: %v", err)
+func (r *RequestStat) initSketch() error {
+	latencies := protocols.SketchesPool.Get()
+	if latencies == nil {
+		return errors.New("error recording kafka transaction latency: could not create new ddsketch")
```

Comment:
Add the log to the caller as done over here:
https://github.com/DataDog/datadog-agent/pull/34080/commits/b16fd2ea8b6a61aaeac20622012c5a449f0958b5